apps/ejabberd/src/mod_mongoose_admin.erl

```diff
@@ -0,0 +1,234 @@
+      {security_policy, [user]},
+      {args, [{caller, binary}, {to, binary}, {msg, binary}]},
+      {result, ok}
+     ],
+     [
+      {name, get_messages},
+      {category, <<"messages">>},
+      {desc, <<"Get recent messages">>},
+      {module, ?MODULE},
+      {function, get_recent_messages},
+      {action, read},
+      {security_policy, [user]},
+      {args, [{caller, binary}, {other, binary}, {limit, integer}]},
+      {result, []}
+     ],
+     [
+      {name, change_password},
+      {category, <<"users">>},
+      {desc, <<"Change user password">>},
+      {module, ?MODULE},
+      {function, change_user_password},
+      {action, update},
+      {security_policy, [user]},
+      {identifiers, [host, user]},
+      {args, [{host, binary}, {user, binary}, {newpass, binary}]},
+      {result, ok}
+     ]
+    ].
+
+kick_session(Host, User, Resource) ->
+    J = jid:make(User, Host, Resource),
+    ejabberd_sm:route(
+      jid:make(<<"">>, <<"">>, <<"">>),
+      J,
+      {broadcast, {exit, <<"kicked">>}}),
+    <<"kicked">>.
+
+list_sessions(Host) ->
+    Lst = ejabberd_sm:get_vh_session_list(Host),
+    [jid:to_binary(JID) || {JID, _, _, _} <- Lst].
+
+registered_users(Host) ->
+    Users = ejabberd_auth:get_vh_registered_users(Host),
+    SUsers = lists:sort(Users),
+    [jid:to_binary(US) || US <- SUsers].
+
+register(Host, User, Password) ->
+    case ejabberd_auth:try_register(User, Host, Password) of
+        ok ->
+            list_to_binary(io_lib:format("User ~s@~s successfully registered", [User, Host]));
+        {error, exists} ->
+            String = io_lib:format("User ~s@~s already registered at node ~p",
+                                   [User, Host, node()]),
+            throw({exists, String});
+        {error, Reason} ->
+            String = io_lib:format("Can't register user ~s@~s at node ~p: ~p",
+                                   [User, Host, node(), Reason]),
+            throw({error, String})
+    end.
+
+unregister(Host, User) ->
+    ejabberd_auth:remove_user(User, Host),
+    <<"">>.
+
+send_message(From, To, Body) ->
+    Packet = build_packet(message_chat, Body),
+    F = jid:from_binary(From),
+    T = jid:from_binary(To),
+    ejabberd_hooks:run(user_send_packet,
+                       F#jid.lserver,
+                       [F, T, Packet]),
+    % privacy check is missing, but is it needed?
+    ejabberd_router:route(F, T, Packet),
+    <<"">>.
+
+registered_commands() ->
+    [#{name => mongoose_commands:name(C),
+       category => mongoose_commands:category(C),
+       action => mongoose_commands:action(C),
+       desc => mongoose_commands:desc(C)
+      } || C <- mongoose_commands:list(admin)].
+
+get_recent_messages(Caller, Other, Limit) ->
+    Res = lookup_recent_messages(Caller, Other, Limit),
+    lists:map(fun record_to_map/1, Res).
+
+change_user_password(Host, User, Password) ->
+    ejabberd_auth:set_password(User, Host, Password).
+
+record_to_map({Id, From, Msg}) ->
```

Comment:
The name is a bit missleading and I didn't improve that when making the PR.
It is not Erlang's record it's rather an `item` returned by `mod_mam`.